app/rest/proxy/avatar_store.go

```diff
@@ -85,3 +101,41 @@ func (fs *FSAvatarStore) location(id string) string {
 	partition := checksum64 % 100
 	return path.Join(fs.storePath, fmt.Sprintf("%02d", partition))
 }
+
+// Resizes an image of supported format (PNG, JPG, GIF) to the size of "limit" px of the biggest side
+// (width or height) preserving aspect ratio.
+// Returns original reader if resizing is not needed or failed.
+func resize(reader io.Reader, limit int) (io.Reader, error) {
```

Comment:
returned error feels wrong to me as this is not really an error and caller forced to do an unusual thing - to use return value even if error != nil. I think it will be better to return just the reader and log directly inside resize fn.